tools/autograd/derivatives.yaml

```diff
@@ -1234,6 +1234,21 @@
   self: slice_backward_wrapper(grad, self.sizes(), dim, start, end, step)
   result: auto_linear
 
+- name: slice_scatter(Tensor self, Tensor mutated_view, int dim=0, int? start=None, int? end=None, int step=1) -> Tensor
+  self: grad * slice_scatter(at::ones(self.sizes(), self.options()), at::zeros(mutated_view.sizes(), mutated_view.options()), dim, start, end, step)
```

Comment:
Why not just slice_scatter zeros into grad?

Reply:
cc @zou3519 on use of `self.options()` here which seems questionable

Reply:
The use of self.options() here is probably OK. We don't have an invariant that "if `self` is a BatchedTensor, then its gradient should also be a BatchedTensor"; if it's necessary the gradient will get expanded later down the line

Reply:
slice_scatter directly into grad makes sense, thanks